src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -231,47 +231,46 @@ internal ItemManager(RecentItemsManager recentItemsManager, IGlobalOptionService
                 // to `MatchResult` to achieve this.
                 initialListOfItemsToBeIncluded.Sort(MatchResult<VSCompletionItem>.SortingComparer);
 
-                var showCompletionItemFilters = _globalOptions.GetOption(CompletionViewOptions.ShowCompletionItemFilters, document?.Project.Language);
+                var filteringResult = initialRoslynTriggerKind == CompletionTriggerKind.Deletion
+                        ? HandleDeletionTrigger(reason, initialListOfItemsToBeIncluded, filterText, hasSuggestedItemOptions)
+                        : HandleNormalFiltering(GetFilterMethod(), filterText, filterReason, data.Trigger.Character, initialListOfItemsToBeIncluded, hasSuggestedItemOptions);
+
+                if (!filteringResult.HasValue)
+                    return null;
 
+                var (selectedItemIndex, selectionHint, uniqueItem) = filteringResult.Value;
+                var useAggressiveDefaultsMatching = session.TextView.Options.GetOptionValue<bool>(AggressiveDefaultsMatchingOptionName);
```

Comment:
this feels *very* weird.  we don't generally go back to text views to pick up options.  Is there a reason this is doner this way, instead of using any of hte other normal roslyn option mechanisms?

Reply:
for example, why is this not using _globalOptions like the code below?

Reply:
This is an option defined in editor, I didn't know if we could use _globalOptions for this.